tests/test_interface.py

```diff
@@ -233,6 +234,46 @@ def test_tab_completion(cls):
         assert regex.match(name) is not None
 
 
+_STATUS_PRINT_IGNORES = {
+    '.AttenuatorCalculatorBase',
+    '.BadSlitPositionerBase',
+    '.DelayBase',
+    '.IPM_Det',
+    '.InOutPVStatePositioner',
+    '.KappaXYZStage',
+    '.PVPositionerComparator',
+    '.PVPositionerDone',
+    '.PVPositionerIsClose',
+    '.PseudoSingleInterface',
+    '.PulsePicker',
+    '.SlitsBase',
+    '.SyncAxesBase',
+}
+
+
+def _should_check_status_prints(cls):
+    """Filter out classes for checking ``status_info``."""
+    fully_qualified_name = f'{cls.__module__}.{cls.__name__}'
+    if any(name in fully_qualified_name for name in _STATUS_PRINT_IGNORES):
+        return False
+
+    # Otherwise, include any Devices that inherit from BaseInterface.
+    return BaseInterface in cls.mro()
+
+
+@pytest.mark.parametrize(
+    'cls',
+    [pytest.param(cls, id=f'{cls.__module__}.{cls.__name__}')
+     for cls in conftest.find_all_device_classes()
+     if _should_check_status_prints(cls)
+     ]
```

Comment:
Can we sort this list (and the one for tab completion) so that the test readout is easier to parse? It looks like this now:
```
...
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.sample_delivery.M3BasePLCDevice] PASSED [ 52%]
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.sample_delivery.PCM] PASSED [ 53%]
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.spectrometer.VonHamosCrystal] PASSED [ 53%]
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.gon.Kappa] PASSED [ 53%]
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.state.StateRecordPositionerBase] PASSED [ 53%]
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.lens.LensStackBase] PASSED [ 53%]
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.lodcm.H1N] PASSED [ 53%]
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.timetool.TimetoolWithNav] PASSED [ 53%]
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.lens.LensStack] PASSED [ 53%]
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.pim.PPM] PASSED [ 54%]
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.state.CombinedStateRecordPositioner] PASSED [ 54%]
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.lxe._ReversedTimeToolDelay] PASSED [ 54%]
tests/test_interface.py::test_smoke_status_prints[pcdsdevices.attenuator.AttenuatorCalculator_AT2L0] PASSED [ 54%]
...
```

Reply:
This annoyed me and I (apparently) made an attempt to fix it, but didn't realize it was doing the equivalent of `return list(set(sorted(modules)))` before. Fixed for good in ce3512d .

Reply:
Looks :+1: :+1: to me!